modules/banner/react/lib/Banner.tsx

```diff
@@ -6,7 +6,7 @@ import {SystemIcon} from '@workday/canvas-kit-react-icon';
 import {exclamationCircleIcon, exclamationTriangleIcon} from '@workday/canvas-system-icons-web';
 import {focusRing} from '@workday/canvas-kit-react-common';
 
-export interface BannerProps extends React.HTMLAttributes<HTMLButtonElement> {
+export interface BannerProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
```

Comment:
Ah. Now I understand what you mean. I guess I assumed `HTMLAttributes<HTMLButtonElement>` would automatically apply attributes from `HTMLButtonElement`, but I see that it doesn't.

Reply:
I think we should use or make a lint rule to help with this. I see a pattern where the Element type should match the React element interface.

Reply:
@NicholasBoll Not all React Elements match the interface for that element. `HTMLAttributes<HTMLSpanElement>` is a good example... it's not `SpanHTMLAttributes<HTMLSpanElement>`.

It can be a lint rule or maybe we can make an indexable type so that it's maintained in one place? Something like:

```
// shared/types.ts

interface ElementProps {
  body: ...,
  br: ...,
  button: React.ButtonHTMLAttributes<HTMLButtonElement>,
  canvas: ...
  ...
}

// banner/react/Banner.tsx

type ButtonElementProps = ElementProps['button'];
export interface BannerProps extends ButtonElementProps {
  ...
}